keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -106,14 +107,64 @@ For the first implementation we do not want to support restore in the
 outside of Kubernetes. The restore is a container engine only operation
 in this first step.
 
-The forensic use case is targeted to be part of the next (1.24) release.
-
 Although this KEP only adds checkpointing support to the kubelet the CRI API in
 the corresponding code pull request is extended to support *checkpoint* and
 *restore* in the CRI API. The reason to add *restore* to the CRI API without
 implementing it in the kubelet is to make development and especially testing
 easier on the container engine level.
 
+#### CRI Updates
+
+The CRI API will be extended to introduce two new RPC APIs:
+```
+    // CheckpointContainer checkpoints a container
+    rpc CheckpointContainer(CheckpointContainerRequest) returns (CheckpointContainerResponse) {}
+    // RestoreContainer restores a container
+    rpc RestoreContainer(RestoreContainerRequest) returns (RestoreContainerResponse) {}
+```
+with the following parameters:
+```
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Location of the checkpoint archive used for export/import
+    string archive_location = 1;
+}
+
+message CheckpointContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+}
+
+message CheckpointContainerRequest {
+    // ID of the container to be checkpointed.
+    string container_id = 1;
+    // Options for checkpointing.
+    CheckpointContainerOptions options = 2;
+}
+
+message CheckpointContainerResponse {
+}
+
+message RestoreContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // ID of the sandbox into which the container should be restored
+    // If this is not specified the container will be restored
```

Comment:
We can drop this and always require the pod sandbox id to be specified.

Reply:
Good point. The comment was actually wrong and no longer was correct. I removed the comment. The pod sandbox ID is indeed always required.